pipenv/core.py

```diff
@@ -607,24 +612,24 @@ def shorten_path(location, bold=False):
 def do_where(virtualenv=False, bare=True):
     """Executes the where functionality."""
     if not virtualenv:
-        location = project.pipfile_location
-        # Shorten the virtual display of the path to the virtualenv.
-        if not bare:
-            location = shorten_path(location)
-        if not location:
+        if not project.pipfile_exists:
```

Comment:
`pipfile_location` no longer returns `None`, checking the existence makes more sense.